### PR TITLE
Changed configuration deserialization to use kebab-case.

### PR DESCRIPTION
**Pull request by davidv1992**

### CODE REVIEW

File: CONFIGURATION.md [hunk 1]
```diff
@@ -63,19 +63,19 @@ Note that peers can also be generated from simply a string containing the addres
...
-| spike_threshold | 900 | Amount of time before a clock difference larger than 125ms is considered real instead of a spike in the network. Lower values ensure large errors are corrected faster, but make the client more sensitive to network issues. Value provided is in seconds. |
-| panic_threshold | 1800 | Largest time difference the client is allowed to correct in one go. Differences beyond this cause the client to abort synchronization. Value provided is in seconds, set to 0 to disable checking of jumps. |
-| startup_panic_threshold | Disabled | Largest time difference the client is allowed to correct during startup. By default, this is unrestricted as we may be the initial source of time for systems without a hardware backed clock. Value provided is in seconds, set to 0 to disable checking of jumps. |
+| min-intersection_survivors | 1 | Minimum number of servers that need to agree on the true time from our perspective for synchronization to start. |
```
Comment (djc) [inline CONFIGURATION.md:66]: Missed one here.

File: CONFIGURATION.md [hunk 2]
```diff
@@ -63,19 +63,19 @@ Note that peers can also be generated from simply a string containing the addres
...
+| min-cluster_survivors | 3 | Number of servers beyond which we do not try to exclude further servers for the purpose of improving measurement precision. Do not change unless familiar with the NTP algorithms. |
```
Comment (djc) [inline CONFIGURATION.md:67]: And here...

File: CONFIGURATION.md [hunk 3]
```diff
@@ -63,19 +63,19 @@ Note that peers can also be generated from simply a string containing the addres
...
+| frequency-tolerance | 15 | Estimate of the short-time frequency precision of the local clock, in parts-per-million. The default is usually a good approximation. |
+| distance-threshold | 1 | Maximum delay to the clock representing ground truth via a peer for that peer to be considered acceptable, in seconds. |
+| frequency-measurement_period | 900 | Amount of time to spend on startup measuring the frequency offset of the system clock, in seconds. Lowering this means the clock is kept actively synchronized sooner, but reduces the precision of the initial frequency estimate, which could result in lower stability of the clock early on. |
```
Comment (djc) [inline CONFIGURATION.md:70]: Here

File: CONFIGURATION.md [hunk 4]
```diff
@@ -63,19 +63,19 @@ Note that peers can also be generated from simply a string containing the addres
...
+| spike-threshold | 900 | Amount of time before a clock difference larger than 125ms is considered real instead of a spike in the network. Lower values ensure large errors are corrected faster, but make the client more sensitive to network issues. Value provided is in seconds. |
+| panic-threshold | 1800 | Largest time difference the client is allowed to correct in one go. Differences beyond this cause the client to abort synchronization. Value provided is in seconds, set to 0 to disable checking of jumps. |
+| startup-panic_threshold | Disabled | Largest time difference the client is allowed to correct during startup. By default, this is unrestricted as we may be the initial source of time for systems without a hardware backed clock. Value provided is in seconds, set to 0 to disable checking of jumps. |
```
Comment (djc) [inline CONFIGURATION.md:73]: And here.

File: CONFIGURATION.md (hunk 1 above)
Comment (davidv1992) [reply CONFIGURATION.md]: fixed

File: CONFIGURATION.md (hunk 2 above)
Comment (davidv1992) [reply CONFIGURATION.md]: fixed

File: CONFIGURATION.md (hunk 3 above)
Comment (davidv1992) [reply CONFIGURATION.md]: fixed

File: CONFIGURATION.md (hunk 4 above)
Comment (davidv1992) [reply CONFIGURATION.md]: fixed

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
I think there are still a few underscore instances left in `CONFIGURATION.md`.

**Review by folkertdev — APPROVED** [review]

**Pull request merged**